File: Data Structure/Heap/LET_no703_Kth Largest Element in a Stream.py
class KthLargest:

    def __init__(self, k: int, nums: List[int]):
        self.temp = []
        self.size = k
        for i in nums:
            heapq.heappush(self.temp, i)
        while len(self.temp) > self.size:
            heapq.heappop(self.temp)

    def add(self, val: int) -> int:
        heapq.heappush(self.temp, val)
        if len(self.temp) > self.size:
                heapq.heappop(self.temp)
        return self.temp[0]

# Your KthLargest object will be instantiated and called as such:
# obj = KthLargest(k, nums)
# param_1 = obj.add(val)

File: Data Structure/Heap/LET_no703_Kth Largest Element in a Stream.cpp
class KthLargest {
private:
    priority_queue<int> pq;
    int size;
public:
    KthLargest(int k, vector<int>& nums) {
        size = k;
        for (int num : nums)
            pq.push(-num);
    }
    
    int add(int val) {
        pq.push(-val);
        while (pq.size() > size)
            pq.pop();
        return -pq.top();
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */